src/wallet/storage.rs

```diff
@@ -157,3 +157,103 @@ impl FileData {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bip39::Mnemonic;
+    use bitcoin::Network;
+    use bitcoind::tempfile::tempdir;
+
+    #[test]
+    fn test_write_and_read_wallet_to_disk() {
+        let temp_dir = tempdir().unwrap();
+        let file_path = temp_dir.path().join("test_wallet.json");
+        let mnemonic = Mnemonic::generate(12).unwrap().to_string();
+
+        let original_wallet_store = WalletStore::init(
+            "test_wallet".to_string(),
+            &file_path,
+            Network::Bitcoin,
+            mnemonic.clone(),
+            "passphrase".to_string(),
+        )
+        .unwrap();
+
+        original_wallet_store.write_to_disk(&file_path).unwrap();
+
+        let read_wallet = WalletStore::read_from_disk(&file_path).unwrap();
+
+        assert_eq!(read_wallet.wallet_name, original_wallet_store.wallet_name);
+        assert_eq!(read_wallet.network, original_wallet_store.network);
+        assert_eq!(
+            read_wallet.external_index,
+            original_wallet_store.external_index
+        );
+        assert_eq!(
+            read_wallet.incoming_swapcoins,
+            original_wallet_store.incoming_swapcoins
+        );
```

Comment:
Stupid me, I missed that you can derive `ParitalEq, Eq` for `WalletStore`. This whole assert series could be just 
`assert_eq!(read_wallet, original_wallet_store);`. 

Including all the asserts below.

Reply:
My bad... I forgot to ask...